src/yunohost/app.py

```diff
@@ -1153,36 +1157,9 @@ def app_makedefault(operation_logger, app, domain=None):
 
     operation_logger.related_to.append(("domain", domain))
 
-    if "/" in app_map(raw=True)[domain]:
-        raise YunohostValidationError(
-            "app_make_default_location_already_used",
-            app=app,
-            domain=app_domain,
-            other_app=app_map(raw=True)[domain]["/"]["id"],
-        )
-
     operation_logger.start()
 
-    # TODO / FIXME : current trick is to add this to conf.json.persisten
-    # This is really not robust and should be improved
-    # e.g. have a flag in /etc/yunohost/apps/$app/ to say that this is the
-    # default app or idk...
-    if not os.path.exists("/etc/ssowat/conf.json.persistent"):
-        ssowat_conf = {}
-    else:
-        ssowat_conf = read_json("/etc/ssowat/conf.json.persistent")
```

Comment:
I'm wondering what to do with the existing setups ... or maybe we just warn people about it in the next release note

Reply:
I was thinking about a migration but I don't know if it's right (you have to run it manually anyway...) Is there some mecanism to run some code after a yunohost package upgrade?

Reply:
Oh didn't know about the legacy utils yet!
Since we have to get rid of the old redirect rules when ssowatconf is run(?), we might as well try to get the redirection right
I'll give it a go